packages/core/src/auth/tokens/token-scope.ts

```diff
@@ -1,26 +1,68 @@
 export enum CacheRole {
   None = 'none',
   ReadWrite = 'readwrite',
+  ReadOnly = 'readonly',
+}
+
+export class All {}
+export class CacheName {
+  name: string;
+  constructor(name: string) {
+    this.name = name;
+  }
+}
+export type CacheSelector = All | CacheName;
```

Comment:
what would you think about supporting `| string` here and then having the code implicitly convert that to a `CacheName`?

Reply:
I can do that. Curious why that is better? Is it so that you can do something like CacheSelector('foo')?

Reply:
I'm kind of just guessing because I don't think we have a place in the PR where you show examples of what user code will look like, but I think it will allow something like:

`new CachePermission(ReadOnly, 'my-awesome-cache')` instead of `new CachePermission(ReadOnly, new CacheName('my-awesome-cache')`.

and actually now that I think about it I think maybe CacheName should be an interface instead of a class. Maybe we can zoom about this for a second tomorrow morning?

Reply:
Sure, will ping you tomorrow morning

Reply:
Here are a couple of integration tests that show user code: https://github.com/momentohq/client-sdk-javascript/pull/598/files#diff-ae2e3efaec41bcc9a06f77ab73967c608830d8c711127a11f30253ca462a3fe3
I was waiting to test these with the server before I write more.

Reply:
Changed CacheName and TopicName to interface instead of class. Also added string as a member of the CacheSelector and TopicSelector unions.